### PR TITLE
Implementation/3 implement product service

### PR DESCRIPTION
**Pull request by Sem4enko**

### CODE REVIEW

File: src/main/java/team/asd/service/ProductService.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		if (CollectionUtils.isNotEmpty(list) && list.contains(null)) {
+			throw new WrongProductException("Product cannot be null");
+		}
+	}
+
+	private long countProductByState(List<IsProduct> productList, ProductState state) throws WrongProductException {
+		if (CollectionUtils.isEmpty(productList)) {
+			return 0L;
+		}
+		checkProducts(productList);
```
Comment (rdovgan) [inline src/main/java/team/asd/service/ProductService.java]: No need to make validation n times. Move this to outer method. Also, you can change implementation using Streams

**Review by rdovgan — CHANGES_REQUESTED** [review]
Improve performance and change methods to use StreamAPI `groupBy` and `collect` methods

File: src/main/java/team/asd/service/ProductService.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	@Override
+	public List<IsProduct> defineProductsWithCreatedState(List<IsProduct> productList) {
+		if (CollectionUtils.isEmpty(productList)) {
+			return Collections.emptyList();
+		}
+
+		return productList.stream()
+				.filter(Objects::nonNull)
+				.filter(element -> element.getState()
+						.equals(Created))
```
Comment (NckGrs) [inline src/main/java/team/asd/service/ProductService.java]: please do not use static imports for enums
https://carlosbecker.com/posts/avoid-static-imports/

Comment (rdovgan) [reply src/main/java/team/asd/service/ProductService.java]: `Created.equals(element.getState())`

**Review by rdovgan — APPROVED** [review]

**Pull request merged**